contracts/Pool.sol

```diff
@@ -190,14 +192,58 @@ abstract contract Pool {
         }
         receiver.lastFundsPerCycle = lastFundsPerCycle;
         receiver.nextCollectedCycle = collectedCycle;
-        if (collected > 0) transferToSender(uint128(collected));
+        transferToSender(uint128(collected));
+    }
+
+    /// @notice Updates all the sender parameters of the sender of the message.
+    ///
+    /// Tops up and withdraws unsent funds from the balance of the sender.
+    ///
+    /// Sets the target amount sent on every block from the sender of the message.
+    /// On every block this amount is rounded down to the closest multiple of the sum of the weights
+    /// of the receivers and proxies and split between them proportionally to their weights.
+    /// Each receiver and proxy then receives their part from the sender's balance.
+    /// If set to zero, stops funding.
+    ///
+    /// Sets the weight of the provided receivers and proxies of the sender of the message.
+    /// The weight regulates the share of the amount sent on every block
+    /// that each of the sender's receivers and proxies get.
+    /// Setting a non-zero weight for a new receiver or
+    /// a new proxy adds it to the list of the sender's receivers.
+    /// Setting zero as the weight for a receiver or a proxy
+    /// removes it from the list of the sender's receivers.
+    /// @param topUp The topped up amount
+    /// @param withdrawAmt The amount to be withdrawn, must not be higher than available funds.
+    /// Can be `WITHDRAW_ALL` to withdraw everything.
+    /// @param amountPerBlock The target amount to be sent on every block.
+    /// Can be `AMOUNT_PER_BLOCK_UNCHANGED` to keep the amount unchanged.
+    /// @param updatedReceivers The list of the updated receivers and their new weights
+    /// @param updatedProxies The list of the updated proxies and their new weights
+    function updateSenderInternal(
+        uint128 topUp,
+        uint128 withdrawAmt,
+        uint128 amountPerBlock,
+        ReceiverWeight[] calldata updatedReceivers,
+        ReceiverWeight[] calldata updatedProxies
+    ) internal suspendPayments returns (uint128 withdrawn) {
+        topUpInternal(topUp);
+        withdrawn = withdrawInternal(withdrawAmt);
+        setAmountPerBlockInternal(amountPerBlock);
+        setReceiversInternal(updatedReceivers, updatedProxies);
     }
 
     /// @notice Must be called when funds have been transferred into the pool contract
     /// in order to top up the message sender
     /// @param amount The topped up amount
-    function onTopUp(uint128 amount) internal suspendPayments {
-        senders[msg.sender].startBalance += amount;
+    function topUpSuspending(uint128 amount) internal suspendPayments {
```

Comment:
Could you update the documentation for this function?

Reply:
What changes would you expect to see? The function didn't change its behavior or API except the name.

Reply:
I just find the “must be called” portion confusing. Maybe just plainly describing what it does:

> Adds the given amount to the senders pool balance and recalculates the funding period.

Reply:
Fixed